File: problem29-divideTwoIntegers/divideTwoIntegers.py
"""
@name divideTwoIntegers 
@desc A function that divides two integers 
@return quotient 
"""
def divideTwoIntegers(dividend, divisor):
    sum = 0
    counter = 0
    negativeFlag = False
    if (dividend < 0 and divisor > 0):
        negativeFlag = True
    elif (divisor < 0 and dividend > 0):
        negativeFlag = True
    if (dividend < 0):
        dividend = dividend * -1
    if (divisor < 0):
        divisor = divisor * -1
  
    while (dividend >= sum + divisor):
        sum = sum + divisor
        counter = counter + 1
    if (negativeFlag):
        counter = counter * -1
    print(counter)
    return counter
divideTwoIntegers(10,3)

File: problem29-divideTwoIntegers/divideTwoIntegers.js
/**
 * @name divideTwoIntegers 
 * @desc A function that divides two integers without using any multiplication and divisions 
 * @param dividend - 
 * @param divisor - 
 * @return quotient - The resulting quotient integer
 */
function divideTwoIntegers(dividend, divisor) {
    let sum = 0;
    let counter = 0;
    let negativeFlag = false;
    if (dividend < 0 && divisor > 0) {
        negativeFlag = true;
    } else if (divisor < 0 && dividend > 0) {
        negativeFlag = true;
    }
    if (dividend < 0) {
        dividend = dividend * -1;
    }
    if (divisor < 0) {
        divisor = divisor * -1;
    }
    while (dividend >= sum + divisor) {
        sum = sum + divisor
        console.log("Sum: " + sum)
        counter++;
        console.log("Counter: " + counter)
    }
    if (negativeFlag) {
        counter = counter * -1;
    }
 
    console.log("quotient: " + counter)
    return counter;
}
divideTwoIntegers(-7,3)